src/tvdb/client.py

```diff
@@ -0,0 +1,192 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Literal, overload, override
+
+import aiohttp
+from yarl import URL
+
+from src.settings import TVDB_API_KEY
+from src.tvdb.generated_models import (
+    MovieBaseRecord,
+    MovieExtendedRecord,
+    SearchResult,
+    SeriesBaseRecord,
+    SeriesExtendedRecord,
+)
+from src.tvdb.models import MovieResponse, SearchResponse, SeriesResponse
+from src.utils.log import get_logger
+
+log = get_logger(__name__)
+
+type JSON_DATA = dict[str, JSON_DATA] | list[JSON_DATA] | str | int | float | bool | None  # noice
+
+
+class _Media(ABC):
+    def __init__(self, client: "TvdbClient"):
+        self.client = client
+        self._id: int | None = None
+
+    @property
+    def id(self) -> int | str | None:
+        return self._id
+
+    @id.setter
+    def id(self, value: int | str) -> None:
+        self._id = int(str(value).split("-")[1])
+
+    def __call__(self, media_id: str) -> "_Media":
+        self.id = media_id
+        return self
+
+    @abstractmethod
+    async def search(self, search_query: str, limit: int = 1) -> list[Any]: ...
+
+    @abstractmethod
+    async def fetch(self, *, extended: bool = False) -> Any: ...
+
+
+class Series(_Media):
+    """Class to interact with the TVDB API for series."""
+
+    @override
+    async def search(self, search_query: str, limit: int = 1) -> list[SearchResult]:
+        """Search for a series in the TVDB database.
+
+        :param search_query:
+        :param limit:
+        :return:
```

Comment:
Either write something about these params, or don't mention them at all, there's no point in just listing them in the docstring without anything. (Applies for all such occasions)

Reply:
Yeah that is a wip, I was planning on documenting everything.